File: back/GatewayAPI/src/main/java/gatewayapi/repository/offer/OfferRepository.java
package gatewayapi.repository.offer;

import dto.offerapi.Offer;
import dto.offerapi.OfferInput;
import gatewayapi.client.OfferClient;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class OfferRepository {

    private final OfferClient offerClient;

    public OfferRepository(OfferClient offerClient) {
        this.offerClient = offerClient;
    }

    public List<Offer> findAll() {
        return offerClient.getAllOffers();
    }

    public List<Offer> findActive() {
        return offerClient.getActiveOffers();
    }

    public Offer findById(int id) {
        return offerClient.getOfferById(id);
    }

    public Offer create(OfferInput offerInput) {
        return offerClient.createOffer(offerInput);
    }

    public Offer update(int id, OfferInput offerInput) {
        return offerClient.updateOffer(id, offerInput);
    }

    public void delete(int id) {
        offerClient.deleteOffer(id);
    }
}

File: front/MyApplication/core/domain/src/main/java/defalt/domain/repository/impl/OfferRepository.kt
package defalt.domain.repository.impl

import defalt.domain.datasource.offer.IOfferRemoteDataSource
import defalt.domain.entity.offer.Offer
import defalt.domain.entity.offer.OfferInput
import defalt.domain.repository.service.IOfferRepository
import defalt.utils.NetworkResult

class OfferRepository(
    private val remoteDataSource: IOfferRemoteDataSource,
) : IOfferRepository {



    override suspend fun getActiveOffers(): NetworkResult<List<Offer>> =
        remoteDataSource.getActiveOffers()



    override suspend fun getOffers(): NetworkResult<List<Offer>> =
        remoteDataSource.getOffers()

    override suspend fun getOfferById(id: Int): NetworkResult<Offer> =
        remoteDataSource.getOfferById(id)



    override suspend fun createOffer(offerInput: OfferInput): NetworkResult<Offer> =
        remoteDataSource.createOffer(offerInput)

    override suspend fun updateOffer(id: Int, offerInput: OfferInput): NetworkResult<Offer> =
        remoteDataSource.updateOffer(id, offerInput)

    override suspend fun deleteOffer(id: Int): NetworkResult<Unit> =
        remoteDataSource.deleteOffer(id)
}
